manifests/istio-control/istio-discovery/templates/deployment.yaml

```diff
@@ -66,7 +81,7 @@ spec:
           - {{ .Release.Namespace }}
 {{- end }}
 
-{{- if and .Values.global.controlPlaneSecurityEnabled }}
+{{- if and .Values.global.controlPlaneSecurityEnabled .Values.global.istiod.enabled }}
```

Comment:
istiod.enabled flag is gone now

Reply:
Ah, will suck when backporting. Fixed in master.